codec/encode_test.go

```diff
@@ -210,6 +211,26 @@ func TestEncodeFieldValue_Group(t *testing.T) {
 	}
 }
 
+func marshalDeterministic(m proto.Message) ([]byte, error) {
```

Comment:
The change to this test is totally superfluous. The underlying type, `*testprotos.Test`, is _known_ to implement this interface, so we don't need to use a type assertion and don't need a fallback at all.

Reply:
The generated proto code as it is checked-in to the repository today does implement the interface, but once you re-generate it with the protobuf api-v2 branch, it won’t anymore.

Reply:
I'd rather keep the code simple and intuitive until it's necessary to change. AFAICT, there is still a good bit of time before v2 becomes GA (e.g. an actual "v2.0.0" tag) so I don't think there's much risk in leaving it. Please revert.

Reply:
One more patch to carry internally, then :(

Done.

Reply:
> One more patch to carry internally, then :(

I don't quite follow. Are you suggesting that this code is somehow preventing the v2 branch from dropping support for this old interface? That seems a bit implausible given how many projects there are (including many private ones) that will also need that patch to be carried forward.

Reply:
No, I’m saying in our code base we use api-v2, and in order to use the protoreflect package, we need to carry a few patches. I was hoping that we could share these changes with everyone (which should make everyone’s transition to v2 easier, eventually) and avoid maintaining custom patches.

Reply:
And what does the custom patch look like? This line is in a test and is referring to a method that exists statically in the same codebase. How could this possibly impact consumers of protoreflect?

Reply:
We throw away protoreflect’s generated files and re-generate them with our version of protoc in our BUILD files.

Reply:
Why? The only protos in the protoreflect repo are in `internal/testprotos` and only used in protoreflect tests.

Reply:
We want to ensure the protoreflect code works with our version of protoc, so it seems prudent to use that version everywhere.